CBinding/CLang/CLangManager.cs

```diff
@@ -0,0 +1,482 @@
+﻿using System;
+using CBinding;
+using ClangSharp;
+using MonoDevelop.Ide;
+using MonoDevelop.Projects;
+using System.Collections.Generic;
+using System.Threading;
+using MonoDevelop.Ide.Editor;
+using System.Runtime.InteropServices;
+using CBinding.Refactoring;
+using CBinding.Parser;
+using System.IO;
+
+namespace CBinding
+{
+	/// <summary>
+	/// Class to manage clang Translation units, thread safe
+	/// </summary>
+	public class CLangManager : IDisposable
+	{
+		public readonly object syncroot = new object ();
+		private CProject project;
+		private CXIndex index;
+		private Dictionary<string, CXTranslationUnit> translationUnits;
+
+		public Dictionary<string, CXTranslationUnit> TranslationUnits {
+			get {				
+				lock (syncroot) {
+					return translationUnits;
+				}
+			}
+		}
+
+		/// <summary>
+		/// Constructor
+		/// </summary>
+		/// <param name="proj">
+		/// A <see cref="CProject"/> reference: project which the manager manages
+		/// </param>
+		public CLangManager (CProject proj)
+		{
+			project = proj;
+			index = clang.createIndex (0, 0);
+			translationUnits = new Dictionary<string, CXTranslationUnit> ();
+			project.DefaultConfigurationChanged += CompilerArgumentsUpdate;
```

Comment:
Do you need to unsubscribe this when disposing so it doesn't leak? If this has the same lifetime as the project, it should be fine as-is, just checking.